### PR TITLE
Implement network selection

### LINKED ISSUE #4682: Implement network selection and remove hardcoded IPs
### Description

We have hard coded Lisk Service IP in 10 different places, yet most hooks dynamically retrieve and use it. We should drop the hardcoded ones in favour of the dynamically retrieved and stored value. 
Additionally, said process is quite hard to follow. It appears that we have technical debt to pay.

### Acceptance criteria

- Should able to select supported networks
- Should handle multiple service URLs supported for same application
- Should handle the application and network error gracefully
- Implement the proposed UI, [Light mode](https://www.figma.com/file/KcrDpvWEKQhdGwNd4CZ5NY/Desktop-Prototype?node-id=3643%3A249687&t=tr6ynawLvEWJNpga-4) and Dark mode

- The application should bootstrap from the selected network (default network is mainnet)
  - Use `settings` and add field `mainchainNetwork`, default it will be `undefined` once the user selects a network update the settings `mainchainNetwork` to selected value.

- Hardcode the list of known networks and its ips
  - On bootstrapping and network selection, filter the networks that are unavailable i.e not 200

- Implement the [UI changes](https://www.figma.com/file/KcrDpvWEKQhdGwNd4CZ5NY/Desktop-Prototype?node-id=3643%3A253175&t=tr6ynawLvEWJNpga-4) to network selection
- The bootstrap flow:
    - Initially the application will fetch the settings value `mainchainNetwork`, the default mainchain name is `Lisk` and hardcode the `Lisk` logo.
    - Probe the mainchain status URL, `baseURL/network/status` and continue user to `dashboard` based on success. i.e `http://devnet-service.liskdev.net:9901/api/v3/network/status`
    - Once user lands on dashboard, fetch the selected mainchain onchain data i.e `https://alphanet-service.liskdev.net/api/v3/blockchain/apps` and also fetch metadata for all the onchain apps by passing chainID `https://alphanet-service.liskdev.net/api/v3/blockchain/apps/meta?chainID`
    - Update the redux by merging both onchain and offchain information

#### Switching network when on sidechain/lisk app
- When switching the network when on sidechain, the user should be redirected to the select network flow.
- When switching the network when on mainchain, i.e Application is `Lisk` and network is `Mainnet` and switching to `Testnet` doesn't require to redirect

### Additional information

With regards to environment configurations, we should consider the following:

Lisk supports following networks, devnet, alphanet, betanet, testnet and mainnet. For all of these Lisk Service provides following URLs:

https://devnet-service.liskdev.net:9901
https://alphanet-service.liskdev.net
https://betanet-service.lisk.com
https://testnet-service.lisk.com
https://mainnet-service.lisk.com

Based on the above base URL, we can fetch application metadata and corresponding application specific service URLs.

So in env config file, we should use appropriate network name e.g `devnet` during development, `alphanet` once we tag alpha and release it for testing, `betanet` once we tag beta and release it for testing, etc.

### PR DESCRIPTION
**Pull request by eniolam1000752**
### What was the problem?

This PR resolves #4682

### How was it solved?

- [x] Implement network selection page and functionalities
- [x] Updated functionality on Application Bootstrap to factor in new network selection flow
- [x] write/update unit test

### How was it tested?

Jest
Visually

### CODE REVIEW

File: setup/react/app/ApplicationBootstrap.js [hunk 1]
```diff
@@ -1,45 +1,61 @@
...
   const [isFirstTimeLoading, setIsFirstTimeLoading] = useState(true);
-
+  const [, setCurrentApplication] = useCurrentApplication();
   useTransactionUpdate();
 
+  const {
+    data: networksMainChainStatus,
+    isLoading: isVerifyingNetworkOptions,
+    isError: isErrorVerifyingNetworkOptions,
+  } = useGetNetworksMainChainStatus({ options: { enable: !!mainChainNetwork } });
```
Comment (oskarleonard) [inline setup/react/app/ApplicationBootstrap.js]: `enabled`

File: src/modules/blockchainApplication/manage/hooks/queries/useBlockchainApplicationMeta.js [hunk 2]
```diff
@@ -38,15 +37,18 @@ export const useBlockchainApplicationMeta = ({
...
   };
 
   return useCustomInfiniteQuery({
     keys: [BLOCKCHAIN_APPS_META],
     config,
     client,
-    options,
+    options: {
+      ...options,
+      enable: !!network,
```
Comment (oskarleonard) [inline src/modules/blockchainApplication/manage/hooks/queries/useBlockchainApplicationMeta.js]: `enabled`

File: src/modules/common/components/customRoute/index.js [hunk 3]
```diff
@@ -19,9 +20,15 @@ const CustomRoute = ({ path, exact, isPrivate, forbiddenTokens, component, t, hi
   const { isMigrated } = useCheckLegacyAccount(currentAccount?.metadata?.pubkey);
   const { search = '' } = history.location;
   const { accounts } = useAccounts();
+  const { mainChainNetwork } = useSettings('mainChainNetwork');
 
   Piwik.tracking(history, token);
 
+  if (!mainChainNetwork && path !== routes.selectNetwork.path) {
+    history.replace({ pathname: routes.selectNetwork.path });
+    return <div />;
```
Comment (oskarleonard) [inline src/modules/common/components/customRoute/index.js]: is the div here to prevent flicker?
Why cant we just use `<Redirect` ?

Comment (eniolam1000752) [reply src/modules/common/components/customRoute/index.js]: resolved

File: src/modules/blockchainApplication/manage/hooks/queries/useBlockchainApplicationMeta.js (hunk 2 above)
Comment (eniolam1000752) [reply src/modules/blockchainApplication/manage/hooks/queries/useBlockchainApplicationMeta.js]: 👍

File: setup/react/app/ApplicationBootstrap.js (hunk 1 above)
Comment (eniolam1000752) [reply setup/react/app/ApplicationBootstrap.js]: 👍

File: src/modules/blockchainApplication/manage/hooks/useGetMergedApplication.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    isLoading: isLoadingAppsMetaData,
+    error: errorGettingAppsMetaData,
+    refetch: refetchAppsMetaData,
+  } = useBlockchainApplicationMeta({
+    config: { params: { chainID: params.chainID, network: networkName } },
+    client: metaDataClient,
+    options: { enabled: isEnabled },
+  });
+
+  const isBlockchainExporeEnabled =
```
Comment (oskarleonard) [inline src/modules/blockchainApplication/manage/hooks/useGetMergedApplication.js]: missing l in expLore
`isBlockchainExploreEnabled`

File: src/modules/blockchainApplication/manage/hooks/useGetMergedApplication.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+    return applicationsMetadata.map(({ chainID, ...restApplication }) => ({
+      chainID,
+      ...restApplication,
+      ...applications.find((app) => app.chainID === chainID),
+    }));
+  }, [applicationsMetadata, applications]);
+
+  const retry = () => {
+    if (hasError) {
```
Comment (oskarleonard) [inline src/modules/blockchainApplication/manage/hooks/useGetMergedApplication.js]: We probably need:
`if (hasError && isBlockchainExporeEnabled) {`
To prevent this "error"
image

Since refetch doesnt respect the `enabled` option.

Comment (eniolam1000752) [reply src/modules/blockchainApplication/manage/hooks/useGetMergedApplication.js]: would be deprecated by https://github.com/LiskHQ/lisk-desktop/pull/4859/#issue-1631188237

File: src/modules/blockchainApplication/manage/hooks/useGetMergedApplication.js (hunk 4 above)
Comment (eniolam1000752) [reply src/modules/blockchainApplication/manage/hooks/useGetMergedApplication.js]: would be deprecated by https://github.com/LiskHQ/lisk-desktop/pull/4859/#issue-1631188237

File: src/modules/network/components/networkSwitcherDropdown/NetworkSwitcherDropdown.js [hunk 6]
```diff
@@ -62,29 +65,24 @@ function NetworkSwitcherDropdown({ noLabel, onLoadApplications, onLoadNetworkSta
...
+    const isSuccess =
+      blockchainAppsMetadata.isSuccess &&
+      networkStatus.isSuccess &&
+      !blockchainAppsMetadata.isFetching &&
+      !networkStatus.isFetching;
+    if (isSuccess) {
+      onNetworkSwitchSuccess?.(true);
+    } else {
+      onNetworkSwitchSuccess?.(false);
+    }
```
Comment (eniolam1000752) [inline src/modules/network/components/networkSwitcherDropdown/NetworkSwitcherDropdown.js]: this can be 

`onNetworkSwitchSuccess?.(isSuccess);`

File: src/modules/network/components/selectNetwork/SelectNetwork.test.js [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    Object.keys(networks)
+      .filter((networkKey) => networks[networkKey].isAvailable)
+      .forEach((networkKey) => {
+        if (networks[networkKey] && networkKey !== networkKeys.devnet) {
+          expect(screen.getByText(networks[networkKey].label)).toBeTruthy();
+        }
+      });
+  });
+
+  it('should be possible to click Continue to dashboard button if isSuccess and !isFetching', () => {
```
Comment (eniolam1000752) [inline src/modules/network/components/selectNetwork/SelectNetwork.test.js]:
```suggestion
  it('should be possible to click continue to dashboard button if isSuccess and  !isFetching', () => {
```

File: src/modules/network/components/networkSwitcherDropdown/NetworkSwitcherDropdown.js (hunk 6 above)
Comment (oskarleonard) [reply src/modules/network/components/networkSwitcherDropdown/NetworkSwitcherDropdown.js]: Nice, it will even be more KIS :D

**Review by oskarleonard — APPROVED** [review]

**Review by ManuGowda — APPROVED** [review]

**Comment by eniolam1000752** [comment]
LGTM
